app/src/main/java/org/p2p/wallet/home/ui/main/HomePresenter.kt

```diff
@@ -103,22 +97,26 @@ class HomePresenter(
         }
     }
 
-    private data class ViewState(
-        val tokens: List<Token.Active> = emptyList(),
-        val ethereumState: EthereumHomeState = EthereumHomeState(),
-        val visibilityState: VisibilityState = VisibilityState.Hidden,
-        val username: Username? = null,
-        val areZerosHidden: Boolean,
-    ) {
-        val ethereumTokens: List<Token.Eth>
-            get() = ethereumState.ethereumTokens
-        val ethereumBundleStatuses: Map<String, List<ClaimStatus?>>
-            get() = ethereumState.ethereumBundleStatuses
+    override fun attach(view: HomeContract.View) {
+        super.attach(view)
+        launch {
+            tokensPolling.shareTokenPollFlowIn(this).onEach {
+                val isRefreshing = it.first.isEmpty()
+                view.showRefreshing(isRefreshing)
+            }.collect {
+                val solTokens = it.first
```

Comment:
val (solTokens, ethTokens) = it

Reply:
can be just        }.collect { (solTokens, ethTokens) ->